tests/EpisodeIntegrationServiceTests/ReceiveDataTests/ReceiveDataTests.cs

```diff
@@ -0,0 +1,69 @@
+using Moq;
+using Microsoft.Extensions.Logging;
+using NHS.ServiceInsights.Common;
+using System.Text;
+
+namespace NHS.ServiceInsights.EpisodeManagementService;
+
+[TestClass]
+public class ReceiveDataTests
+{
+    private readonly Mock<IHttpRequestService> _mockHttpRequestService = new();
+    private readonly Mock<ILogger<ReceiveData>> _mockLogger = new();
+    private readonly ReceiveData _function;
+    public ReceiveDataTests()
+    {
+        Environment.SetEnvironmentVariable("ProcessDataURL", "ProcessDataURL");
+
+        _function = new ReceiveData(_mockLogger.Object, _mockHttpRequestService.Object);
+    }
+
+    [TestMethod]
+    public async Task Run_ShouldLogValidJsonAndCallSendPost()
+    {
+        // Arrange
+        var validJson = "{\"name\":\"John Doe\"}";
+        var stream = new MemoryStream(Encoding.UTF8.GetBytes(validJson));
+
+        // Act
+        await _function.Run(stream, "test.json");
+
+        // Assert
+        _mockLogger.Verify(log =>
+            log.Log(
+            LogLevel.Information,
+            0,
+            It.Is<object>(state => state.ToString().Contains("JSON is valid.")),
+            null,
+            (Func<object, Exception, string>)It.IsAny<object>()),
+            Times.Once);
+
+        _mockHttpRequestService.Verify(x => x.SendPost(It.IsAny<string>(), It.IsAny<string>()), Times.Once);
```

Comment:
In our verifications of the SendPost method calls we can also check that the URL is what we expect